submissions/UA-girl/task_friends_app/js/app.js

```diff
@@ -0,0 +1,354 @@
+            if (state.toggled) {
+                this.sortCards(state.direction, filter);
+            }
+        }
+    }
+
+    resetFilterState(updateGender = false) {
+        this.filterStates.age = {
+            toggled: false,
+            age: null,
+        };
+        if (updateGender) {
+            this.filterStates.gender = {
+                toggled: false,
+                gender: null,
+            }
+        }
+    }
+
+    resetSortingState() {
+        this.sortingStates = {
+            name: {
+                toggled: false,
+                direction: null,
+            },
+            registration: {
+                toggled: false,
+                direction: null
+            },
+            age: {
+                toggled: false,
+                direction: null,
+            }
+        };
+    }
+
+    genderFilter(gender, wasToggled=true) {
+        if (wasToggled) {
+            this.filterStates.gender.gender = gender;
+            this.restoreFromFilter();
+        } else {
+            this.filterStates.gender.toggled = true;
+            this.filterStates.gender.gender = gender;
+            this.cards = this.cards.filter(person => person.gender === gender);
+        }
+    }
+
+    ageFilter(maxAge) {
+        this.filterStates.age.toggled = true;
+        document.querySelector('.label-range').innerHTML = `Max age: ${maxAge}`;
+        if(this.filterStates.age.age < +maxAge) {
+            this.filterStates.age.age = +maxAge;
+            this.restoreFromFilter();
+        } else {
+            this.filterStates.age.age = +maxAge;
+            this.cards = this.cards.filter(person => person.age <= + maxAge);
+        }
+    }
+
+    sortCards(value, filterName) {
+        switch (filterName) {
+            case 'age':
+                this.sortByAge(value);
+                break;
+            case 'name':
+                this.sortByName(value);
+                break;
+            case 'registration':
+                this.sortByRegistrationDate(value)
+        }
+    }
+
+    nameFilter(name) {
+        name = name.trim().toLowerCase();
+        this.restoreFromFilter();
+        this.cards = this.cards.filter(person => person.name.first.toLowerCase().startsWith(name));
+    }
+
+    sortByAge(isFromLowest) {
+        this.resetSortingState();
+        this.cards.sort((person1, person2) => person1.ageCompare(person2));
+        this.sortingStates.age.toggled = true;
+        this.sortingStates.age.direction = isFromLowest;
+        if (isFromLowest === 'down') {
+            this.cards.reverse()
+        }
+    }
+
+    sortByName(isFromLowest) {
+        this.resetSortingState();
+        this.cards.sort((person1, person2) => person1.nameCompare(person2));
+        this.sortingStates.name.toggled = true;
+        this.sortingStates.name.direction = isFromLowest;
+        if (isFromLowest === 'down') {
+            this.cards.reverse()
+        }
+    }
+
+    sortByRegistrationDate(isFromLowest) {
```

Comment:
Functions `sortByAge`, `sortByName` and `sortByRegistrationDate` have a lot of similar logic. Consider some universal function. It will help you to make your code more drier. More info about the DRY principle you can find [here](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself#:~:text=The%20DRY%20principle%20is%20stated,their%20book%20The%20Pragmatic%20Programmer)